pkg/operator2/operator.go

```diff
@@ -400,6 +403,9 @@ func (c *authOperator) handleSync(operatorConfig *operatorv1.Authentication) err
 		return err
 	}
 
+	proxyConfig := c.handleProxyConfig()
+	resourceVersions = append(resourceVersions, proxyConfig.GetResourceVersion())
```

Comment:
It annoys me that this adds another thing that needs explicit RV tracking.  Oh well.

Use `proxyConfig.ResourceVersion` directly - the generic methods should only be used when dealing with Object Meta generically, not when you have a specific type.

Reply:
I guess most of it could be replaced with the config-observation logic as done in {K,O}AS-O, we'd probably only be left with checking the resource versions of configmaps/secrets (for CA/cert+priv key updates)